src/base_object-inl.h

```diff
@@ -123,6 +123,22 @@ BaseObject::MakeLazilyInitializedJSTemplate(Environment* env) {
   return t;
 }
 
+template <int Field>
+void BaseObject::InternalFieldGet(
+    v8::Local<v8::String> property,
+    const v8::PropertyCallbackInfo<v8::Value>& info) {
+  info.GetReturnValue().Set(info.This()->GetInternalField(Field));
+}
+
+template <int Field, bool (v8::Value::* typecheck)() const>
+void BaseObject::InternalFieldSet(v8::Local<v8::String> property,
+                                  v8::Local<v8::Value> value,
+                                  const v8::PropertyCallbackInfo<void>& info) {
+  // This could be e.g. value->IsFunction().
+  CHECK_IMPLIES(typecheck != nullptr, ((*value)->*typecheck)());
```

Comment:
Since `typecheck` is never nullptr currently, maybe change the CHECK_IMPLIES into a simple CHECK?

You could also remove the default template argument and have a second, single-arg InternalFieldSet() without a check.

Reply:
Yeah, I’ve turned it into a simple `CHECK` and removed the `nullptr` default. We can always adjust this if we want.